src/Network/GRPC/MQTT/Client.hs

```diff
@@ -244,69 +274,111 @@ mqttRequest MQTTGRPCClient{..} baseTopic (MethodName method) useBatchedStream re
+      let errMsg = Text.pack (displayException e)
       logErr mqttLogger errMsg
       return $ MQTTError errMsg
 
-{- | Helper function to run an 'ExceptT RemoteError' action and convert any failure
- to an 'MQTTResult'
--}
-exceptToResult :: (Functor m) => ExceptT RemoteError m (MQTTResult streamtype response) -> m (MQTTResult streamtype response)
+-- | Helper function to run an 'ExceptT RemoteError' action and convert any failure
+-- to an 'MQTTResult'
+--
+-- @since 1.0.0
+exceptToResult ::
+  Functor f =>
+  ExceptT RemoteError f (MQTTResult s rsp) ->
+  f (MQTTResult s rsp)
 exceptToResult = fmap (either fromRemoteError id) . runExceptT
 
-{- | Manages the control signals (Heartbeat and Terminate) asynchronously while
- the provided action performs a request
--}
+-- | Manages the control signals (Heartbeat and Terminate) asynchronously while
+-- the provided action performs a request
+--
+-- @since 1.0.0
 withControlSignals :: (AuxControl -> IO ()) -> IO a -> IO a
 withControlSignals publishControlMsg = withMQTTHeartbeat . sendTerminateOnException
   where
     withMQTTHeartbeat :: IO a -> IO a
-    withMQTTHeartbeat action =
-      withAsync
+    withMQTTHeartbeat action = do
+      Async.withAsync
         (forever (publishControlMsg AuxControlAlive >> sleep heartbeatPeriodSeconds))
         (const action)
 
     sendTerminateOnException :: IO a -> IO a
     sendTerminateOnException action =
       action `onException` publishControlMsg AuxControlTerminate
 
-{- | Connects to the MQTT broker and creates a 'MQTTGRPCClient'
- NB: Overwrites the '_msgCB' field in the 'MQTTConfig'
--}
-connectMQTTGRPC :: (MonadIO io) => Logger -> MQTTGRPCConfig -> io MQTTGRPCClient
+-- | Connects to the MQTT broker and creates a 'MQTTGRPCClient'
+-- NB: Overwrites the '_msgCB' field in the 'MQTTConfig'
+--
+-- @since 1.0.0
+connectMQTTGRPC :: MonadIO io => Logger -> MQTTGRPCConfig -> io MQTTGRPCClient
 connectMQTTGRPC logger cfg = do
-  resultChan <- newTChanIO
-
-  let clientMQTTHandler :: MessageCallback
-      clientMQTTHandler =
-        SimpleCallback $ \_client topic mqttMessage _props -> do
-          logDebug logger $ "clientMQTTHandler received message on topic: " <> unTopic topic <> " Raw: " <> decodeUtf8 mqttMessage
-          atomically $ writeTChan resultChan mqttMessage
-
-  MQTTGRPCClient
-    <$> connectMQTT cfg{_msgCB = clientMQTTHandler}
-    <*> pure resultChan
-    <*> new
-    <*> pure logger
-    <*> pure (fromIntegral (mqttMsgSizeLimit cfg))
-
-disconnectMQTTGRPC :: (MonadIO io) => MQTTGRPCClient -> io ()
-disconnectMQTTGRPC = liftIO . normalDisconnect . mqttClient
-
--- | Returns a 'GRPCIOTimeout' error if the supplied function takes longer than the given timeout.
-grpcTimeout :: (MonadUnliftIO m) => TimeoutSeconds -> m (MQTTResult streamType response) -> m (MQTTResult streamType response)
-grpcTimeout timeLimit action = fromMaybe timeoutError <$> timeout (timeLimit * 1000000) action
+  chan <- liftIO newTChanIO
```

Comment:
Why is the `liftIO` necessary now?

Reply:
`newTChanIO` was pulling from `relude`'s re-export `unliftio-core` I believe